File: 1979-find-greatest-common-divisor-of-array/1979-find-greatest-common-divisor-of-array.py
class Solution:
    def Gcd(this,a,b):
        while b!=0:
            a,b = b,a%b
        return abs(a)
    def findGCD(self, nums: List[int]) -> int:
        a = min(nums)
        b = max(nums)
        c=  self.Gcd(a,b)
        return c

File: 1979-find-greatest-common-divisor-of-array/1979-find-greatest-common-divisor-of-array.java
class Solution {
    public int Gcd(int a , int b){
        while (b!=0){
            int temp = a;
            a= b;
            b=temp%b;
        }
        return Math.abs(a);
    }
    public int findGCD(int[] nums) {
        int mn =Integer.MAX_VALUE;
        int mx = 0;
        for(int i=0;i<nums.length;i++){
            if(nums[i]>mx){
                mx = nums[i];
            }
             if(mn>nums[i]){
                mn = nums[i];
            }
        }
        return Gcd(mn,mx);
    }
}